p2p/libp2p/topicProcessors.go

```diff
@@ -0,0 +1,77 @@
+package libp2p
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/p2p"
+)
+
+type topicProcessors struct {
+	processors    map[string]p2p.MessageProcessor
+	identifiers   []string
```

Comment:
why do we need a list of identifiers since we already have the map keys?
I think it can be safely removed, as for getList() you can create the identifiers list together with the processors list from the map.

Reply:
I wanted to call the processors in the exact same order as the processors were registered. Of course this is not mandatory, it just gives a more deterministic flavor to the process. The list was introduced to not sort each time the map keys. Not having a strong opinion on keeping the list. :)

Reply:
removed identifiers, added comment on networkMessenger.RegisterMessageProcessor that states the order is not guaranteed.